kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/LogReplay.java

```diff
@@ -165,6 +172,11 @@ public long getVersion() {
     return logSegment.getVersion();
   }
 
+  /** Returns the crc info for the current snapshot if the checksum file is read */
+  public Optional<CRCInfo> getCurrentCrcInfo() {
```

Comment:
is there any reason that we only save it if it's the current version? in the future won't we also want to still save the info for an earlier one? (to pass along as much info as possible)

Reply:
current snapshot is sufficient for unblocking simple crc write, so I only expose the currently version. We could incrementally expose more if needed in the future.